### DISCUSSION TITLE
Regarding kubelogin what is the path of the token? What is the validity and can this token to be modified?

### DISCUSSION THREAD

**Discussion by amruta53** [Azure/kubelogin#302] [Q&A]
In the linux machine, we are not able to find the token in ${HOME}/.kube/cache/kubelogin folder
What is the validity of this token, can it be modified?
Can we generate and configure the token on our own?

**Comment by weinong** [comment]
whether a token is cached in local filesystem is determined by the login mode you use. 
from [this doc](https://azure.github.io/kubelogin/cli/remove-tokens.html), 

> Note that only devicelogin, interactive, and ropc login modes will cache the token.

what do you mean configuring the token on your own though? if you already have the AAD token, you don't need kubelogin. you can do `kubectl --token=${MY_OWN_TOKEN}`
